src/test/java/org/openmrs/contrib/qaframework/automation/InpatienttSteps.java

```diff
@@ -0,0 +1,103 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.uitestframework.test.TestData;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+public class InpatienttSteps extends Steps {
+
+	private TestData.PatientInfo testPatient;
+
+	@Before(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void visitHomePage() {
+		testPatient = createTestPatient();
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_ENCOUNTER)
+	public void destroy() {
+		deletePatient(testPatient);
+		quit();
+	}
+
+	@Given("a user clicks on Find Patient app from Home page")
+	public void goToFindPatientRecordPage() {
+		findPatientPage = homePage.goToFindPatientRecord();
+	}
+
+	@Then("the system loads find patient record page")
+	public void systemloadsFindPatientRecordPage() {
+		assertTrue(textExists("Find Patient Record"));
+	}
+
+	@When("a user searches an existing patient from find patient record page")
+	public void searchExistingPatient() {
+		findPatientPage.enterPatient(testPatient.identifier);
+		findPatientPage.waitForPageToLoad();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+
+	@And("a user starts visit from the patient dashboard page")
+	public void clickOnStartVisit() {
+		visitsDashboardPage = dashboardPage.startVisit();
+	}
+
+	@Then("the system loads patient visit dashboard page")
+	public void systemloadsPatientVisitDashboardPage() {
+		assertTrue(visitsDashboardPage.containsText("Visits"));
+	}
+
+	// Admission encounter
```

Comment:
What value does the above comment add?

Reply:
> What value does the above comment add?

Very interesting question !

Well it has no effect when building the code base. However, I thought it would be of value to include some important comments for  current and future technical devs to know where each user story is actually implemented. As we have three user stories for encounters ie [Admission](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/resources/features/refapp-2.x/08-encounter/inpatient.feature#L13), [Transfer](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/resources/features/refapp-2.x/08-encounter/inpatient.feature#L17) and [Discharge](https://github.com/openmrs/openmrs-contrib-qaframework/blob/534d5e28cffa213f003ca12a625eff0795d47958/src/test/resources/features/refapp-2.x/08-encounter/inpatient.feature#L21) I felt compelled to do add comment in the selenium implementation so that in case any of the user story is failing in future it lessen the geeks time to figure out its equivalent selenium implementation.

Reply:
Can't you do that in the method name itself?

Reply:
Also, what do you think about this? https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/

Reply:
> Can't you do that in the method name itself?

This is more than an opinion but a great idea. Am now a consultant to myself and qualify to guide others in this context.

> Also, what do you think about this? https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/

Have kept the resource in my archive for future reference.  Its precise and brief with a well defined content embedded!